W-16139934 AEH sso restructuring and azure content jh (#49)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/troubleshoot-sso-errors.adoc

A 400 error occurs when logging in to the portal with SSO if the external identity provider application doesn’t have the proper redirect URIs defined.

> To troubleshoot this issue, ensure to include all of the necessary redirect URIs:

Reviewer: ```suggestion
To troubleshoot this issue, make sure to include necessary redirect URIs:
```

. Navigate to *Anypoint Platform*, enter your username and password, and  click *Sign in*.
. From Anypoint Platform, select *API Experience Hub* from the list of products. 

> . Go to From the *Manage your API portal* page, click *Preview and publish your portal*.

Reviewer: Read aloud...

. Click *Publish*. 

== API Carousel Not Loading

After logging in to the portal successfully, you can't see APIs in the API Carousel.

A few issues can cause this error to occur:

* The groups claim is missing in the identity provider's application configuration.
+ 

> To troubleshoot this issue, add the claim in the application and ensure at least one group is being sent.

Reviewer: ```suggestion
To troubleshoot this issue, add the claim in the application and make sure at least one group is being sent.
```

Reviewer: Style change coming: Salesforce prefers "make sure" to "ensure".

. From *Provider Name*, select the name of the corresponding Salesforce identity provider.
. From *Type*, select *Member* and click *Add*.

> * The *Auth provider Default Scopes* field is not configured properly.

Reviewer: ```suggestion
* The *Auth provider Default Scopes* field isn’t configured properly.
```

+
To troubleshoot this issue, ensure the *Default Scopes* field has the *profile openid email groups* value.

. In *Salesforce*, go to *Setup*.
. In the *Quick Find* box, enter `Auth`, and then select *Auth Providers*.
. Click *Edit*.
. In the *Auth Provider Detail* section, enter `profile openid email groups`.
. Click *Save*.

== Can't Log In
When logging in to the portal with an identity provider, you see a *Problem Logging In* error that the third-party identifier can’t be found. An incorrect portal URL can cause issues logging in.

To troubleshoot this issue, ensure that the URL is correct in the configuration. For Okta, the `/userinfo` part of the URL path must be lowercase.

== Single Sign-on Error

> After logging on, you see a *Single Sign-On Error* in the portal. Issues can occur when the SSO or the Salesforce Auth. Provider configuration is incorrect or missing necessary information. 

Reviewer: ```suggestion
After logging in, *Single Sign-On Error* shows in the portal. Issues can occur if the SSO or the Salesforce Auth. Provider configuration is incorrect or missing necessary information. 
```

Reviewer: Is it really called "Salesforce Auth. Provider"?

Author: yep

A few issues can cause this error to occur:

* The Entity ID isn't the same in both the Salesforce Auth. Provider and the identity provider.
+
To troubleshoot this issue for SAML protocol configurations:
+
. Ensure the *Entity ID* is the same in both the Salesforce Auth. Provider's *Single Sign-On Settings* page and the identity provider. 
+
. Click *SAML Assertion Validation* in the *Single Sign-On Settings* page to check for errors in the assertion.

> * Making changes to *Single Sign-On Settings* page without uploading the SSO certificate. 

Reviewer: ```suggestion
* You change the *Single Sign-On Settings* page without uploading the SSO certificate. 
```

Reviewer: (Not a complete sentence; not parallel with other bullets.)

> * Tou make changes to *Single Sign-On Settings* page without uploading the SSO certificate. 

Reviewer: ```suggestion
* You make changes to *Single Sign-On Settings* page without uploading the SSO certificate. 
```

+

> To troubleshoot this issue, ensure you upload the SSO certificate each time you make changes to the *Single Sign-On Settings* page in Salesforce.

Reviewer: ```suggestion
To troubleshoot this issue, ensure you upload the SSO certificate each time you change the *Single Sign-On Settings* page in Salesforce.
```

* The expected claims are missing in the identity provider's application.
+
After logging on using SSO, you see an SSO error with the error details in the URL. An error from the registration handler can cause this error. The most common cause is missing expected claims. 
+
To troubleshoot this issue, ensure you have the expected claims configured in the application. For more information, see xref:gathering-set-up-information-for-sso.adoc# required-claims-and-attributes-for-sso[Required Claims and Attributes for SSO].

== Portal Errors
After logging in to the portal using SSO, you get an error in the portal.

> The following issues can cause this error when the identity provider's application has missing or incorrect information:

Reviewer: ```suggestion
These issues can cause the error if the identity provider's application has missing or incorrect information:
```

* Incorrect URLs
* The member isn't assigned to a user or group in the application

To troubleshoot these issues:

> * Review the URLS in identity provider's application SSO configuration and the *Auth. Provider/Single Sign-On* page from Salesforce > *Setup* to ensure you're using the correct single sign-on URL.

Reviewer: ```suggestion
* Review the URLs in identity provider's application SSO configuration and the *Auth. Provider/Single Sign-On* page from Salesforce *Setup* to make sure you're using the correct single sign-on URL.
```

Reviewer: (Don't use the > path if it's not a simple path.)

* Ensure the member is assigned to a user or a group in the identity provider application.

== Issues Testing the SSO Configuration

> Issues can occur when making changes to the SSO configuration, and then testing the configuration.

Reviewer: ```suggestion
Issues can occur when changing the SSO configuration, and then testing the configuration.
```

> To troubleshoot this issue, ensure you have a clean Salesforce and Anypoint session before testing SSO:

Reviewer: ```suggestion
To troubleshoot this issue, make sure you have a clean Salesforce and Anypoint Platform session before testing SSO:
```

. Log out of the portal, then log in again with the SSO user. 
+
This step ensures the registration handler executes properly.

> . Delete the Anypoint token from the Named Credential.

Reviewer: Anypoint Platform?

+

> This step clears the Anypoint's user session and trigger authentication with the new configuration information to send to Anypoint:

Reviewer: Anypoint Platform?

> This step clears the Anypoint's user session and trigger authentication with the new configuration information to send to Anypoint:

Reviewer: ```suggestion
This step clears the Anypoint Platform user session and triggers authentication with the new configuration information to send to Anypoint Platform:
```
